### ISSUE TITLE
Renovate produces yarn.lock with duplication

### ISSUE THREAD

**Issue by scinos** [renovatebot/renovate#6084]
**What Renovate type are you using?**

GitHub app

**Describe the bug**

After updating a version, Renovate did not deduplicate `yarn.lock`, despite having `postUpdateOptions: yarnDedupeHighest` in our config options.

**Did you see anything helpful in debug logs?**

No :(

**To Reproduce**

Example: https://github.com/Automattic/wp-calypso/pull/41518 (includes Renovate log).

* It updated `@testing-library/react` from `9.3.3` to `9.5.0`. (good)
* As part of this update, it also updated the transitive dep `@types/testing-library__dom` from `^6.0.0` to `^6.12.1` (good)
* It did leave behind `@types/testing-library__dom@*` that can be deduped to `6.14.0` (bug)

Relevant line: https://github.com/Automattic/wp-calypso/pull/41518/files#diff-8ee2343978836a779dc9f8d6b794c3b2R3362

Running `yarn-deduplicate` locally it shows
```
Package "@types/testing-library__dom" wants * and could get 6.14.0, but got 6.12.1
```

**Additional context**

N/A

**Comment by viceice** [comment]
@scinos please provide the debug logs, there should be the dedupe call visible

**Comment by scinos** [comment]
Logs for that branch:
```
DEBUG: processBranch with 1 upgrades(branch="renovate/testing-library-react-9.x")
DEBUG: Setting baseBranch to master(branch="renovate/testing-library-react-9.x")
DEBUG: latest commit(branch="renovate/testing-library-react-9.x")
{
  "branchName": "master",
  "latestCommitDate": "2020-04-28 08:57:45 +0200"
}
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: branchExists=true(branch="renovate/testing-library-react-9.x")
DEBUG: Branch pr rebase requested: false(branch="renovate/testing-library-react-9.x")
DEBUG: Branch has 1 upgrade(s)(branch="renovate/testing-library-react-9.x")
DEBUG: Checking if PR has been edited(branch="renovate/testing-library-react-9.x")
DEBUG: Found existing branch PR(branch="renovate/testing-library-react-9.x")
DEBUG: Checking schedule(at any time, null)(branch="renovate/testing-library-react-9.x")
DEBUG: No schedule defined(branch="renovate/testing-library-react-9.x")
DEBUG: Branch already exists(branch="renovate/testing-library-react-9.x")
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: Branch does not need rebasing(branch="renovate/testing-library-react-9.x")
DEBUG: Using parentBranch: renovate/testing-library-react-9.x(branch="renovate/testing-library-react-9.x")
DEBUG: manager.getUpdatedPackageFiles()(branch="renovate/testing-library-react-9.x")
DEBUG: npm.updateDependency(): dependencies.@testing-library/react = 9.5.0(branch="renovate/testing-library-react-9.x")
DEBUG: No package files need updating(branch="renovate/testing-library-react-9.x")
DEBUG: Getting updated lock files(branch="renovate/testing-library-react-9.x")
DEBUG: lock file dirs(branch="renovate/testing-library-react-9.x")
{
  "dirs": {
    "yarnLockDirs": [],
    "npmLockDirs": [],
    "pnpmShrinkwrapDirs": [],
    "lernaDirs": []
  }
}
DEBUG: Writing repo lerna.json (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing repo .npmrc (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing repo .yarnrc (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing package.json files(branch="renovate/testing-library-react-9.x")
{
  "packageFiles": [
    "apps/full-site-editing/package.json",
    "apps/notifications/package.json",
    "apps/o2-blocks/package.json",
    "apps/wpcom-block-editor/package.json",
    "client/extensions/hello-dolly/package.json",
    "client/extensions/sensei/package.json",
    "client/extensions/woocommerce/package.json",
    "client/extensions/wp-super-cache/package.json",
    "client/extensions/zoninator/package.json",
    "client/lib/formatting/decode-entities/package.json",
    "client/lib/i18n-utils/package.json",
    "client/lib/wp/package.json",
    "client/my-sites/customize/package.json",
    "client/my-sites/jetpack-manage-error-page/package.json",
    "client/my-sites/picker/package.json",
    "client/package.json",
    "package.json",
    "packages/babel-plugin-i18n-calypso/package.json",
    "packages/babel-plugin-transform-wpcalypso-async/package.json",
    "packages/calypso-analytics/package.json",
    "packages/calypso-build/package.json",
    "packages/calypso-codemods/package.json",
    "packages/calypso-color-schemes/package.json",
    "packages/calypso-polyfills/package.json",
    "packages/components/package.json",
    "packages/composite-checkout/package.json",
    "packages/data-stores/package.json",
    "packages/effective-module-tree/package.json",
    "packages/eslint-config-wpcalypso/package.json",
    "packages/eslint-plugin-wpcalypso/package.json",
    "packages/format-currency/package.json",
    "packages/i18n-calypso-cli/package.json",
    "packages/i18n-calypso/package.json",
    "packages/load-script/package.json",
    "packages/material-design-icons/package.json",
    "packages/media-library/package.json",
    "packages/photon/package.json",
    "packages/popup-monitor/package.json",
    "packages/react-i18n/package.json",
    "packages/tree-select/package.json",
    "packages/viewport-react/package.json",
    "packages/viewport/package.json",
    "packages/webpack-config-flag-plugin/package.json",
    "packages/webpack-extensive-lodash-replacement-plugin/package.json",
    "packages/webpack-inline-constant-exports-plugin/package.json",
    "packages/wpcom-proxy-request/package.json",
    "packages/wpcom.js/package.json"
  ]
}
DEBUG: Writing yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: Writing .yarnrc to /mnt/renovate/gh/Automattic/wp-calypso(branch="renovate/testing-library-react-9.x")
DEBUG: Writing any updated package files(branch="renovate/testing-library-react-9.x")
DEBUG: No updated lock files in branch(branch="renovate/testing-library-react-9.x")
DEBUG: No files to commit(branch="renovate/testing-library-react-9.x")
DEBUG: Checking if we can automerge branch(branch="renovate/testing-library-react-9.x")
DEBUG: mergeStatus=no automerge(branch="renovate/testing-library-react-9.x")
DEBUG: Ensuring PR(branch="renovate/testing-library-react-9.x")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/testing-library-react-9.x")
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: Found existing PR(branch="renovate/testing-library-react-9.x")
DEBUG: Processing existing PR(branch="renovate/testing-library-react-9.x")
DEBUG: Pull Request #41518 does not need updating(branch="renovate/testing-library-react-9.x")
DEBUG: Checking #41518 for automerge(branch="renovate/testing-library-react-9.x")
{
  "automerge": false,
  "automergeType": "pr",
  "automergeComment": "automergeComment"
}
DEBUG: No automerge(branch="renovate/testing-library-react-9.x")
```

I don't see any dedupe call

**Comment by viceice** [comment]
Please trigger a rebase and post the logs after retry.

This logs are from a second run and renovate sees an uptodate branch. So nothing to do.

**Comment by scinos** [comment]
These are the new logs. They are different, but still not mention to deduplication:

```
DEBUG: processBranch with 1 upgrades(branch="renovate/testing-library-react-9.x")
DEBUG: Setting baseBranch to master(branch="renovate/testing-library-react-9.x")
DEBUG: latest commit(branch="renovate/testing-library-react-9.x")
{
  "branchName": "master",
  "latestCommitDate": "2020-04-29 07:58:55 +0200"
}
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: branchExists=true(branch="renovate/testing-library-react-9.x")
DEBUG: Branch pr rebase requested: true(branch="renovate/testing-library-react-9.x")
DEBUG: Branch has 1 upgrade(s)(branch="renovate/testing-library-react-9.x")
DEBUG: Checking if PR has been edited(branch="renovate/testing-library-react-9.x")
DEBUG: Found existing branch PR(branch="renovate/testing-library-react-9.x")
DEBUG: Checking schedule(at any time, null)(branch="renovate/testing-library-react-9.x")
DEBUG: No schedule defined(branch="renovate/testing-library-react-9.x")
DEBUG: Branch already exists(branch="renovate/testing-library-react-9.x")
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: Manual rebase requested via PR checkbox for #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Using parentBranch: undefined(branch="renovate/testing-library-react-9.x")
DEBUG: manager.getUpdatedPackageFiles()(branch="renovate/testing-library-react-9.x")
DEBUG: npm.updateDependency(): dependencies.@testing-library/react = 9.5.0(branch="renovate/testing-library-react-9.x")
DEBUG: Updating packageFile content(branch="renovate/testing-library-react-9.x")
DEBUG: Updated 1 package files(branch="renovate/testing-library-react-9.x")
DEBUG: Getting updated lock files(branch="renovate/testing-library-react-9.x")
DEBUG: lock file dirs(branch="renovate/testing-library-react-9.x")
{
  "dirs": {
    "yarnLockDirs": [],
    "npmLockDirs": [],
    "pnpmShrinkwrapDirs": [],
    "lernaDirs": [
      "."
    ]
  }
}
DEBUG: Writing repo lerna.json (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing repo .npmrc (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing repo .yarnrc (/mnt/renovate/gh/Automattic/wp-calypso)(branch="renovate/testing-library-react-9.x")
DEBUG: Writing package.json files(branch="renovate/testing-library-react-9.x")
{
  "packageFiles": [
    "apps/full-site-editing/package.json",
    "apps/notifications/package.json",
    "apps/o2-blocks/package.json",
    "apps/wpcom-block-editor/package.json",
    "client/extensions/hello-dolly/package.json",
    "client/extensions/sensei/package.json",
    "client/extensions/woocommerce/package.json",
    "client/extensions/wp-super-cache/package.json",
    "client/extensions/zoninator/package.json",
    "client/lib/formatting/decode-entities/package.json",
    "client/lib/i18n-utils/package.json",
    "client/lib/wp/package.json",
    "client/my-sites/customize/package.json",
    "client/my-sites/jetpack-manage-error-page/package.json",
    "client/my-sites/picker/package.json",
    "client/package.json",
    "package.json",
    "packages/babel-plugin-i18n-calypso/package.json",
    "packages/babel-plugin-transform-wpcalypso-async/package.json",
    "packages/calypso-analytics/package.json",
    "packages/calypso-build/package.json",
    "packages/calypso-codemods/package.json",
    "packages/calypso-color-schemes/package.json",
    "packages/calypso-polyfills/package.json",
    "packages/components/package.json",
    "packages/composite-checkout/package.json",
    "packages/data-stores/package.json",
    "packages/effective-module-tree/package.json",
    "packages/eslint-config-wpcalypso/package.json",
    "packages/eslint-plugin-wpcalypso/package.json",
    "packages/format-currency/package.json",
    "packages/i18n-calypso-cli/package.json",
    "packages/i18n-calypso/package.json",
    "packages/load-script/package.json",
    "packages/material-design-icons/package.json",
    "packages/media-library/package.json",
    "packages/photon/package.json",
    "packages/popup-monitor/package.json",
    "packages/react-i18n/package.json",
    "packages/tree-select/package.json",
    "packages/viewport-react/package.json",
    "packages/viewport/package.json",
    "packages/webpack-config-flag-plugin/package.json",
    "packages/webpack-extensive-lodash-replacement-plugin/package.json",
    "packages/webpack-inline-constant-exports-plugin/package.json",
    "packages/wpcom-proxy-request/package.json",
    "packages/wpcom.js/package.json"
  ]
}
DEBUG: Writing yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: Writing .yarnrc to /mnt/renovate/gh/Automattic/wp-calypso(branch="renovate/testing-library-react-9.x")
DEBUG: Writing any updated package files(branch="renovate/testing-library-react-9.x")
DEBUG: Writing package.json(branch="renovate/testing-library-react-9.x")
DEBUG: Finding package.json for lerna directory "."(branch="renovate/testing-library-react-9.x")
DEBUG: npmrc file found in repository(branch="renovate/testing-library-react-9.x")
DEBUG: Spawning lerna with yarn to create lock files(branch="renovate/testing-library-react-9.x")
DEBUG: Using lerna version ^3.20.2(branch="renovate/testing-library-react-9.x")
DEBUG: Executing command(branch="renovate/testing-library-react-9.x")
{
  "command": "yarn install --ignore-scripts --ignore-engines --ignore-platform --mutex network:31879"
}
DEBUG: exec completed(branch="renovate/testing-library-react-9.x")
{
  "cmd": "yarn install --ignore-scripts --ignore-engines --ignore-platform --mutex network:31879",
  "seconds": 9,
  "stdout": "yarn install v1.22.4\n[1/4] Resolving packages...\nsuccess Saved lockfile.\nDone in 8.06s.\n",
  "stderr": ""
}
DEBUG: Executing command(branch="renovate/testing-library-react-9.x")
{
  "command": "npx lerna@^3.20.2 bootstrap --no-ci -- --ignore-scripts --ignore-engines --ignore-platform --mutex network:31879"
}
DEBUG: exec completed(branch="renovate/testing-library-react-9.x")
{
  "cmd": "npx lerna@^3.20.2 bootstrap --no-ci -- --ignore-scripts --ignore-engines --ignore-platform --mutex network:31879",
  "seconds": 67,
  "stdout": "yarn install v1.22.4\n[1/4] Resolving packages...\nsuccess Already up-to-date.\nDone in 6.76s.\n",
  "stderr": "npx: installed 703 in 54.91s\nlerna notice cli v3.20.2\nlerna info versioning independent\nlerna info bootstrap root only\n"
}
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")
DEBUG: Updated 36 lock files(branch="renovate/testing-library-react-9.x")
{
  "updatedArtifacts": [
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock",
    "yarn.lock"
  ]
}
DEBUG: 37 file(s) to commit(branch="renovate/testing-library-react-9.x")
DEBUG: Committing files to branch renovate/testing-library-react-9.x(branch="renovate/testing-library-react-9.x")
INFO: Branch updated(branch="renovate/testing-library-react-9.x")
{
  "commitHash": "3983233"
}
DEBUG: Checking if we can automerge branch(branch="renovate/testing-library-react-9.x")
DEBUG: mergeStatus=no automerge(branch="renovate/testing-library-react-9.x")
DEBUG: Ensuring PR(branch="renovate/testing-library-react-9.x")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/testing-library-react-9.x")
DEBUG: getBranchPr(renovate/testing-library-react-9.x)(branch="renovate/testing-library-react-9.x")
DEBUG: findPr(renovate/testing-library-react-9.x, undefined, open)(branch="renovate/testing-library-react-9.x")
DEBUG: Found PR #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Returning from graphql open PR list(branch="renovate/testing-library-react-9.x")
DEBUG: Found existing PR(branch="renovate/testing-library-react-9.x")
DEBUG: Processing existing PR(branch="renovate/testing-library-react-9.x")
DEBUG: PR body changed(branch="renovate/testing-library-react-9.x")
{
  "prTitle": "fix(deps): update dependency @testing-library/react to v9.5.0"
}
DEBUG: updatePr(41518, fix(deps): update dependency @testing-library/react to v9.5.0, body)(branch="renovate/testing-library-react-9.x")
DEBUG: PR updated(branch="renovate/testing-library-react-9.x")
{
  "pr": 41518
}
INFO: PR updated(branch="renovate/testing-library-react-9.x")
{
  "pr": 41518,
  "prTitle": "fix(deps): update dependency @testing-library/react to v9.5.0"
}
DEBUG: Ensuring comment ":warning: Artifact update problem" in #41518 is removed(branch="renovate/testing-library-react-9.x")
DEBUG: Getting comments for #41518(branch="renovate/testing-library-react-9.x")
DEBUG: Found 3 comments(branch="renovate/testing-library-react-9.x")
DEBUG: Checking #41518 for automerge(branch="renovate/testing-library-react-9.x")
{
  "automerge": false,
  "automergeType": "pr",
  "automergeComment": "automergeComment"
}
DEBUG: No automerge(branch="renovate/testing-library-react-9.x")
```

**Comment by rarkins** [comment]
@scinos can you tell me a job ID for when a yarn.lock was created or updated?

**Comment by scinos** [comment]
Looks like that branch was created in #170106550. No mention to deduplication there either.

However, in that same job, I can see that branch `renovate/automattic-color-studio-2.x` was created and I do see a `DEBUG: Performing yarn dedupe highest`

I don't know if this is related but in the "bad" branch (the one without the udpdate) I see:
```
DEBUG: lock file dirs(branch="renovate/testing-library-react-9.x")
{
  "dirs": {
    "yarnLockDirs": [],
    "npmLockDirs": [],
    "pnpmShrinkwrapDirs": [],
    "lernaDirs": [
      "."
    ]
  }
}
```

but in the "good" branch (the one deduped) I see:

```
DEBUG: lock file dirs(branch="renovate/automattic-color-studio-2.x")
{
  "dirs": {
    "yarnLockDirs": [
      "yarn.lock"
    ],
    "npmLockDirs": [],
    "pnpmShrinkwrapDirs": [],
    "lernaDirs": [
      "."
    ]
  }
}
```

**Comment by viceice** [comment]
Strange, i see `DEBUG: File is updated: /mnt/renovate/gh/Automattic/wp-calypso/yarn.lock(branch="renovate/testing-library-react-9.x")` replicated a lot of times 🤔

**Comment by rarkins** [comment]
We may need to review whether we call `lerna bootstrap` or just `yarn` too.

**Comment by rarkins** [comment]
I wonder if this is related to the complexity of this repo layout and therefore difficult to reproduce.

**Comment by scinos** [comment]
Is there anything I can do from my side to help debug this?

**Comment by rarkins** [comment]
@scinos as your source repo is massive it's not practical for reproduction/debugging. If you are able to reproduce the same problem with a minimal repo/config, that would make it possible to progress.

**Comment by HonkingGoose** [comment]
I've moved this from Needs Requirements to Blocked, as @rarkins is waiting on a smaller reproduction.
Maybe we should label this `help wanted` and/or `needs reproduction` so that others are nudged to take a look and help?

**Comment by rarkins** [comment]
I think let's close it if there's only one participant and no reproduction possible after this time

**Comment by falkenhawk** [comment]
@rarkins The behavior seems to be buggy also on our private repo.
Example of a job without dedup: #261268882
which resulted in this PR:
![image](https://user-images.githubusercontent.com/14310995/99296081-8048ab80-2846-11eb-9d2f-931ae52b817b.png)

Example of a job where dedupe worked: #261242776
```
DEBUG: Performing yarn dedupe highest(branch="renovate/ts-node-dev-1.x")
```

**Comment by rarkins** [comment]
Please manually diff the two logs around the place where yarn is run and also provide your full config here. And probably best to open a new issue as this one was already long enough from logs

**Comment by falkenhawk** [comment]
@rarkins it's most probably this issue in my case: https://github.com/renovatebot/renovate/issues/7438
